### ISSUE TITLE
.net Maui

### ISSUE THREAD

**Issue by maestro-oc** [BelledonneCommunications/linphone-xamarin#47]
Hi,

Have you got any plans to make the SDK compatible with .net MAUI?

**Comment by KariKummu** [comment]
Any plans for this?

**Comment by JaxInfinitaz** [comment]
+1 for adding maui support.

**Comment by rifaia** [comment]
> +1 for adding maui support.

Now supports Maui:
https://wiki.linphone.org/xwiki/wiki/public/view/Lib/Getting%20started/Xamarin/

**Comment by lucasbrahm** [comment]
It works with .NET MAUI

**Comment by JaxInfinitaz** [comment]
> It works with .NET MAUI

I get segfault errors at runtime when trying with .net maui, have you gotten it to work?

**Comment by lucasbrahm** [comment]
> > It works with .NET MAUI
> 
> I get segfault errors at runtime when trying with .net maui, have you gotten it to work?

I successfully converted my whole Xamarin project with Linphone to .NET MAUI (despite some bugs of .NET MAUI).

What version of Linphone SDK are you using?
Android or iOS?
What does it make crash? Initialize linphone? Start call?

**Comment by pkozak2** [comment]
@lucasbrahm Can you help me? 
My repo: [linphone-maui](https://github.com/pkozak2/linphone-maui)

I initializing linphone, then I want to disable opus codec by user interface. In original Xamarin version - this is working. But in MAUI - AudioPayloadType is always Enabled.
Please check if I initializing Linphone in proper way.

**Comment by headintheclouds21** [comment]
@pkozak2 @lucasbrahm Im also having issues getting Linphone to work  in NET MAUI. Im currently receiving a 'System.DllNotFoundException: linphone' error when trying to create the core. I've downloaded the nuget Linphone SDK and placed the linphonewrapper.cs, llinphonemanager.cs, linphonerc_default, and linphonerc_factory in my NET MAUI project. Maybe I have things in the incorrect place? Any help would be greatly appreciated.

**Comment by lucasbrahm** [comment]
> @lucasbrahm Can you help me? My repo: [linphone-maui](https://github.com/pkozak2/linphone-maui)
> 
> I initializing linphone, then I want to disable opus codec by user interface. In original Xamarin version - this is working. But in MAUI - AudioPayloadType is always Enabled. Please check if I initializing Linphone in proper way.

Looks good to me. Probably a bug with linphone updating or reading codecs.

> @pkozak2 @lucasbrahm Im also having issues getting Linphone to work in NET MAUI. Im currently receiving a 'System.DllNotFoundException: linphone' error when trying to create the core. I've downloaded the nuget Linphone SDK and placed the linphonewrapper.cs, llinphonemanager.cs, linphonerc_default, and linphonerc_factory in my NET MAUI project. Maybe I have things in the incorrect place? Any help would be greatly appreciated.

Check: https://wiki.linphone.org/xwiki/wiki/public/view/Lib/Getting%20started/Xamarin/

If you are going to use nuget, just install linphone SDK nuget to your project. It is not required to copy linphonewrapper.cs and native libraries.

If you are going NOT to use nuget, you have to copy linphonewrapper.cs, and add native libraries (AAR file for Android and all .xcframework for IOS)

**Comment by headintheclouds21** [comment]
@lucasbrahm Thanks for replying. That's helpful to know. I've cloned the NET MAUI project from @pkozak2 that looked good to you, Lucas. I'm still receiving that 'System.DllNotFoundException: linphone' error when trying to create the core. Do you know why that would be? Is it a linphonesdk bug?

**Comment by pkozak2** [comment]
@lucasbrahm can you try disable some codec in your app? For example "opus", then use .enabled() like me to get status? 
I created [ISSUE](https://github.com/BelledonneCommunications/linphone-sdk/issues/378) in sdk repository, but how long I must wait for response - who knows, maybe liblinphone repo will be better place :) 

@headintheclouds21 push your project to github, I will check if its running for me.

**Comment by headintheclouds21** [comment]
@pkozak2 That would be awesome. I'd really appreciate it. I posted my repo here: https://github.com/headintheclouds21/Linphone-Maui I've been trying to mimic your repo to get linphone working in maui. How does mine look to you? I saw the issue you posted. Hopefully there's a response to your it soon :)

**Comment by pkozak2** [comment]
@headintheclouds21 well... this is full copy of my code what I see in commits history. It is working for me without exception.

**Comment by headintheclouds21** [comment]
@pkozak2 yeah i was trying to get your code running on my machine and I keep receiving a **System.DllNotFoundException:** 'linphone' when trying to start the linphonemanager. I'm trying to use the linphone sdk inside my net maui application, but was trying to get yours working before i implemented it into mine. Do you know how you got that dll to be recognized? I figured i must have something in the wrong place. I've been testing using an android emulator if that makes any difference.

**Comment by pkozak2** [comment]
@headintheclouds21 I using VS 2022 in latest version on Windows 11, Android Emulator with API 33. Maybe clear nuget cache? Or try LinphoneSDK.Xamarin from Linphone nuget

**Comment by headintheclouds21** [comment]
@pkozak2 Clearing the nuget cache fixed my issue. Thanks for the helpful tip. Do you know if the Linphone SDK is compatible with iOS or will it only work with Android in NET MAUI? I see there's a xamarin iOS sample for linphone, but not sure if that means it would also be compatible with NET MAUI

**Comment by ParagSaxena** [comment]
@headintheclouds21 Can you please provide working sample for Android for .NET MAUI. Thanks!!
